### PR TITLE
feat: Drawer support classNames

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #43372

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Drawer support classNames to customize build-in module's className |
| 🇨🇳 Chinese | 抽屉组件支持通过 classNames 自定义内置模块的 className  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 12e5a44</samp>

This pull request adds a new `classNames` prop to the `Drawer` and `DrawerPanel` components, which allows users to customize the `className` of the internal regions of the drawer. It also updates the documentation and the demo files for the `drawer` component in both English and Chinese languages. Additionally, it updates the `rc-drawer` dependency to the latest version.

### <samp>🤖 Generated by Copilot at 12e5a44</samp>

*  Add a new `classNames` prop to the `Drawer` and `DrawerPanel` components to allow users to customize the `className` of the internal regions of the drawer ([link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-93e3b026986cf56379143412817a4077a256d276a567a550593b2526310d1a69R6-R13), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-93e3b026986cf56379143412817a4077a256d276a567a550593b2526310d1a69R37), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-93e3b026986cf56379143412817a4077a256d276a567a550593b2526310d1a69R54), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-93e3b026986cf56379143412817a4077a256d276a567a550593b2526310d1a69L70-R86), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-93e3b026986cf56379143412817a4077a256d276a567a550593b2526310d1a69L89-R103), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-93e3b026986cf56379143412817a4077a256d276a567a550593b2526310d1a69L98-R112), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-10ffbf23ed7075c38649dfd0776998a514a56e42e09a91306f89b25e67e07d30L15-R15), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-10ffbf23ed7075c38649dfd0776998a514a56e42e09a91306f89b25e67e07d30R39), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-10ffbf23ed7075c38649dfd0776998a514a56e42e09a91306f89b25e67e07d30R152-R155))
* Update the `rc-drawer` dependency to the latest version that supports the `classNames` prop ([link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L131-R131))
* Add a new code example and documentation for the `classNames` prop in both Chinese and English languages in the `demo` and `index` files of the `drawer` component ([link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-ec0f68188e9dcbb5897e38fee296c9dd30140b14ddc1c0da615eda521b3d978eR1-R7), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-f8e2b6b006447fed670eaa3fde5e356a8f0b615c99c38545829edf34c5a72663R1-R62), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-7855c9de8eab66dde84ba7237f52b0fb88c917e3c8c78ada5fafa1e98c259721R33), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-7855c9de8eab66dde84ba7237f52b0fb88c917e3c8c78ada5fafa1e98c259721R52), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-bb2fbbde54ac7a64e8f607e35d8482d3150594c6466b025b29949ba82b675152R32), [link](https://github.com/ant-design/ant-design/pull/44935/files?diff=unified&w=0#diff-bb2fbbde54ac7a64e8f607e35d8482d3150594c6466b025b29949ba82b675152R51))

### CODE REVIEW

File: components/drawer/DrawerPanel.tsx [hunk 1]
```diff
@@ -3,6 +3,14 @@ import type { DrawerProps as RCDrawerProps } from 'rc-drawer';
...
 import useClosable from '../_util/hooks/useClosable';
 
+export interface DrawerClassNames {
+  header?: string;
+  body?: string;
+  footer?: string;
+  mask?: string;
+  wrapper?: string;
+}
+
```
Comment (crazyair) [inline components/drawer/DrawerPanel.tsx:13]: 这里是单独加的，那 modal 为什么可以引用
image

Comment (kiner-tang) [reply components/drawer/DrawerPanel.tsx:13]: 因为 drawer 的 footer、body、header 都是在 antd 实现的，不是在 rc-drawer 实现的，而 modal 的 footer 、body、header 都是在 rc-dialog 实现的

File: components/drawer/DrawerPanel.tsx [hunk 2]
```diff
@@ -86,7 +119,17 @@ const DrawerPanel: React.FC<DrawerPanelProps> = (props) => {
     }
     const footerClassName = `${prefixCls}-footer`;
     return (
-      <div className={footerClassName} style={footerStyle}>
+      <div
+        className={classNames(
+          footerClassName,
+          drawerClassNames?.footer ?? drawerContext?.classNames?.footer,
```
Comment (MadCcc) [inline components/drawer/DrawerPanel.tsx:125]: 这个得合并一下吧，不能做覆盖
Modal 那边可能漏看了，得一起检查下

Comment (kiner-tang) [reply components/drawer/DrawerPanel.tsx:125]: 好

Comment (kiner-tang) [reply components/drawer/DrawerPanel.tsx]: modal 都是合并的，没有覆盖， drawer 后面漏改了

File: components/drawer/index.tsx [hunk 3]
```diff
@@ -148,6 +153,14 @@ const Drawer: React.FC<DrawerProps> & {
           maskMotion={maskMotion}
           motion={panelMotion}
           {...rest}
+          classNames={{
+            mask: rest.classNames?.mask ?? drawer?.classNames?.mask,
+            content: rest.classNames?.content ?? drawer?.classNames?.content,
```
Comment (MadCcc) [inline components/drawer/index.tsx:158]: 这里和下面的 styles 也需要合并

Comment (kiner-tang) [reply components/drawer/index.tsx]: done

**Review by MadCcc — APPROVED** [review]

File: components/drawer/index.zh-CN.md [hunk 4]
```diff
@@ -45,22 +46,20 @@ demo:
 | --- | --- | --- | --- | --- |
 | autoFocus | 抽屉展开后是否将焦点切换至其 Dom 节点 | boolean | true | 4.17.0 |
 | afterOpenChange | 切换抽屉时动画结束后的回调 | function(open) | - |  |
-| bodyStyle | 可用于设置 Drawer 内容部分的样式 | CSSProperties | - |  |
 | className | Drawer 容器外层 className 设置，如果需要设置最外层，请使用 rootClassName | string | - |  |
+| classNames | 配置抽屉内置模块的 className | `header?: string; body?: string; footer?: string; mask?: string; wrapper?: string;` | - |  |
+| styles | 配置抽屉内置模块的 style | `header?: CSSProperties; body?: CSSProperties; footer?: CSSProperties; mask?: CSSProperties; wrapper?: CSSProperties;` | - | 5.10.0 |
```
Comment (afc163) [inline components/drawer/index.zh-CN.md:51]: 这里是 wrapper，但是代码里是 content。

**Pull request merged**